### PR TITLE
Add kueue jobs to config

### PR DESCRIPTION
**Pull request by kidddddddddddddddddddddd**
This pr adds a config yaml to `config/jobs/kubernetes-sigs/kueue` to  run these tests periodically for kueue.

I'm not familar with both test-infra and kueue. So please point it out if some configuration is wrong.

Here're something that I'm not very clear:
1. `testgrid-alert-email`  for `kueue`?
~~2. I see there is a `kueue-presubmits.yaml` can `testgrid-dashboards` and `testgrid-tab-name` have the same name with tests in `kueue-presubmits.yaml`?~~

### CODE REVIEW

**Comment by kidddddddddddddddddddddd** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/28999/pull-test-infra-prow-checkconfig/1634912120829120512

seems that we can't use `skip_branches` in periodics? 🤔

File: config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml [hunk 1]
```diff
@@ -0,0 +1,253 @@
+periodics:
+  - interval: 1h
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:2]: we can start with 1d

File: config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    name: periodic-kueue-test-unit-main
+    annotations:
+      testgrid-dashboards: sig-scheduling
+      testgrid-tab-name: periodic-kueue-test-unit-main
+      description: "Run kueue unit tests"
+      testgrid-alert-email: release-team@kubernetes.io
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:8]: no alerts, to start

File: config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    decoration_config:
+      timeout: 1h
+    spec:
+      containers:
+        - image: golang:1.19
+          command:
+            - make
+          args:
+            - verify
+  - interval: 1h
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:194]: Maybe we can get rid of this (also for the presubmits). The E2E tests already do a build.

Comment (alculquicondor) [reply config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:194]: Also, follow this discussion https://github.com/kubernetes/test-infra/pull/27411#issuecomment-1466139982

@tenzen-y

**Comment by kidddddddddddddddddddddd** [comment]
Pod can not be created🤔
/retest

File: config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml (hunk 3 above)
Comment (tenzen-y) [reply config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml]: I replied to the above PR.

File: config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml [hunk 4]
```diff
@@ -0,0 +1,245 @@
+periodics:
+  - interval: 1d
+    name: periodic-kueue-test-unit-main
+    annotations:
+      testgrid-dashboards: sig-scheduling
+      testgrid-tab-name: periodic-kueue-test-unit-main
+      description: "Run kueue unit tests"
```
Comment (kannon92) [inline config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:7]:
```suggestion
      description: "Run periodic kueue unit tests"
```

Just to distinguish these tests from the existing unit tests/e2e tests?

**Comment by kidddddddddddddddddddddd** [comment]
Fixed all but https://github.com/kubernetes/test-infra/pull/28999#discussion_r1133917490 based on the comments.

File: config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+        - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230222-b5208facd4-master
+          securityContext:
+            privileged: true
+          command:
+            - runner.sh
+          args:
+            - make
+            - image-local-build
+  - interval: 1d
+    name: periodic-kueue-build-image-release-0-1-and-release-0-2
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:218]: Remove this job. We will not need it.

Comment (kidddddddddddddddddddddd) [reply config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:218]: How about pull-kueue-build-image-main.I see we skip release 0.1/0.2 in that test,should we remove that test too?

Comment (alculquicondor) [reply config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:218]: I'm inclined to remove it. As I said, the E2E tests already do a build, so we are not losing coverage.

File: config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml [hunk 6]
```diff
@@ -0,0 +1,187 @@
+periodics:
+  - interval: 24h
+    name: periodic-kueue-test-unit-main
+    annotations:
+      testgrid-dashboards: sig-scheduling
+      testgrid-tab-name: periodic-kueue-test-unit-main
+      description: "Run periodic kueue unit tests"
+      testgrid-num-columns-recent: '6'
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:8]: this sounds like too little. Maybe 15?

File: config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+      testgrid-num-failures-to-alert: "1"
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:9]: does it matter? given that there is no alerts email

File: config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+      timeout: 1h
+    spec:
+      containers:
+        - image: golang:1.19
+          command:
+            - make
+          args:
+            - test-integration
+  - interval: 24h
+    name: periodic-kueue-test-e2e-main-1-23
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-release-blocking.yaml:58]: Remove this one too. 1.23 is out of support now.
We can add 1.26 in a follow up.

**Comment by alculquicondor** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**